packages/excalidraw/element/flowchart.ts

```diff
@@ -0,0 +1,750 @@
+  elementsMap: ElementsMap,
+  direction: LinkDirection = "right",
+) => {
+  const boundElbowArrows = element.boundElements
+    ?.filter((boundEl) => boundEl.type === "arrow")
+    .map(
+      (boundArrow) => elementsMap.get(boundArrow.id) as ExcalidrawArrowElement,
+    )
+    .filter(
+      (boundArrow) =>
+        boundArrow?.elbowed &&
+        boundArrow.startBinding?.elementId === element.id &&
+        boundArrow.endBinding,
+    );
+
+  if (!boundElbowArrows || boundElbowArrows.length === 0) {
+    return [];
+  }
+
+  const successorsAndHeadingFors = boundElbowArrows.map((boundArrow) => {
+    const successor = elementsMap.get(boundArrow.endBinding!.elementId)!;
+    const headingFor = headingForPointFromElement(
+      element as ExcalidrawBindableElement,
+      aabbForElement(element),
+      [boundArrow.x, boundArrow.y],
+    );
+
+    return {
+      successor,
+      headingFor,
+    };
+  });
+
+  switch (direction) {
+    case "up":
+      return successorsAndHeadingFors
+        .filter(
+          (successorAndHeadingFor) =>
+            successorAndHeadingFor.headingFor[0] === HEADING_UP[0] &&
+            successorAndHeadingFor.headingFor[1] === HEADING_UP[1],
+        )
+        .map((successorAndHeadingFor) => successorAndHeadingFor.successor);
+    case "down":
+      return successorsAndHeadingFors
+        .filter(
+          (successorAndHeadingFor) =>
+            successorAndHeadingFor.headingFor[0] === HEADING_DOWN[0] &&
+            successorAndHeadingFor.headingFor[1] === HEADING_DOWN[1],
+        )
+        .map((successorAndHeadingFor) => successorAndHeadingFor.successor);
+    case "right":
+      return successorsAndHeadingFors
+        .filter(
+          (successorAndHeadingFor) =>
+            successorAndHeadingFor.headingFor[0] === HEADING_RIGHT[0] &&
+            successorAndHeadingFor.headingFor[1] === HEADING_RIGHT[1],
+        )
+        .map((successorAndHeadingFor) => successorAndHeadingFor.successor);
+    case "left":
+      return successorsAndHeadingFors
+        .filter(
+          (successorAndHeadingFor) =>
+            successorAndHeadingFor.headingFor[0] === HEADING_LEFT[0] &&
+            successorAndHeadingFor.headingFor[1] === HEADING_LEFT[1],
+        )
+        .map((successorAndHeadingFor) => successorAndHeadingFor.successor);
+  }
+};
+
+export const getPredecessors = (
+  element: ExcalidrawGenericElement,
+  elementsMap: ElementsMap,
+  direction: LinkDirection = "right",
+) => {
+  // find elbow arrows whose endBinding is the given element
+  const comingInArrows = [...elementsMap.values()]
+    .filter(
+      (el) =>
+        el.type === "arrow" &&
+        el.startBinding &&
+        el.endBinding?.elementId === element.id,
+    )
+    .map((arrow) => elementsMap.get(arrow.id)) as ExcalidrawArrowElement[];
+
+  const predecessorsAndHeadingFors = comingInArrows.map((elbowArrow) => {
+    const predecessor = elementsMap.get(elbowArrow.startBinding!.elementId)!;
```

Comment:
Are we sure the start binding will be there if the user manually deletes the predecessor?

Reply:
> Are we sure the start binding will be there if the user manually deletes the predecessor?

we've filtered above to make sure that we only take elbow arrows whose startBinding is not null